docs/javascript/tutorial.js

```diff
@@ -0,0 +1,28 @@
+var nodeList = $('.progressbar-bubble');
+
+var forwardBtn = document.getElementById('continue');
+
+// Check that forward btn exists on page
+if (forwardBtn) {
+  forwardBtn.addEventListener('onclick', completeState())
+} else {
+  fullState();
+}
+/* Find index of element with active class to fill nodes
+up until index of active class */
+function completeState() {
+  var current = nodeList.filter('.active'),
+    index = nodeList.index(current);
+    
+  for (let i = 0; i < index; i++) {
+    if (index == 0) return
+    nodeList[i].className += 'completed';
+  }
+}
+
+// Fill progress bar when at final, full state
+function fullState() {
+  for (let j = 0; j < nodeList.length; j++) {
+    nodeList[j].className += 'completed';
+  }
+}
```

Comment:
Per our slack convo, this really belongs as logic in jekyll, since the page reloads and the state of the nav can be determined statically.

I haven't used Jekyll a ton, but just glancing at the docs one way you might accomplish this by making the nav an "include".  That is, pulling the nav  stuff out of `ocs/_layouts/tutorials.html`, and [making it an include that takes a parameter](https://jekyllrb.com/docs/includes/#passing-parameters-to-includes).  The parameter would be the "active" page.

Reply:
ACTUALLY…

I think there’s a better pure CSS way…

Check out https://css-tricks.com/child-and-sibling-selectors/#article-header-id-2

and the related selectors.  I think you can style “all siblings that come before”.

If so, you can do this with 1 CSS statement, delete your js file, _and_ not have to mess with jekyll

Reply:
I have trying to play with it but ~ is only for siblings after the element. 
I found this: https://stackoverflow.com/questions/1817792/is-there-a-previous-sibling-css-selector, the one with 190 votes but unsure really how they did that...

Reply:
implementation is a bit longer than expected so I logged an issue for us to come back to: #679

Reply:
I also liked the idea of progress bar being sticky. I logged an issue for this one #681

Reply:
@sigalsax  

> Do not try to find a backward selector...
> That's impossible.
> Instead, only try to realize the truth...
> There is no direction.
> Then you'll see that it is not the CSS that bends
> It is only yourself.

-- [Spoon Boy](https://www.youtube.com/watch?v=uAXtO5dMqEI)

https://jsfiddle.net/cd6w4yfp/

Reply:
bingo bango...

Reply:
that was embarassingly easy... I was thinking about it all wrong. 
ME: "How do I get it to reverse itself..."
Just put normal styling and cover it with `.active ~ .progressbar-bubble`. DUUUUH ^.^

Reply:
#pushed